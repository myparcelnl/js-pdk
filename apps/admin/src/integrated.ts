export * from './common';

export {
  LoadingPageView,
  ModalsView,
  NotificationsView,
  OrderBoxView,
  OrderListItemView,
  PluginSettingsView,
  ProductSettingsView,
  createPdkAdminPlugin,
} from '@myparcel-pdk/frontend-core/src';
