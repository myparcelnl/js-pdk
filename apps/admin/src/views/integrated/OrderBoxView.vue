<template>
  <OrderBox />
</template>

<script lang="ts" setup>
import {toRefs} from 'vue';
import OrderBox from '../OrderBox.vue';
import {extendAdminInstance} from '../../utils';

const props = defineProps({
  orderIdentifier: {
    type: String,
    required: true,
  },
});

const propRefs = toRefs(props);

extendAdminInstance({context: {orderIdentifier: propRefs.orderIdentifier?.value}});
</script>
