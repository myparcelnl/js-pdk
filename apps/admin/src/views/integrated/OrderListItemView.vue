<template>
  <OrderListItem />
</template>

<script lang="ts" setup>
import {toRefs} from 'vue';
import OrderListItem from '../OrderListItem.vue';
import {extendAdminInstance} from '../../utils/extendAdminInstance';

const props = defineProps({
  orderIdentifier: {
    type: String,
    required: true,
  },
});

const propRefs = toRefs(props);

extendAdminInstance({context: {orderIdentifier: propRefs.orderIdentifier?.value}});
</script>
