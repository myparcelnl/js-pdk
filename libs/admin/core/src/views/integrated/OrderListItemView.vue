<template>
  <OrderListItem />
</template>

<script lang="ts" setup>
import OrderListItem from '../OrderListItem.vue';
import {extendAdminInstance} from '../../composables';

const props = defineProps({
  orderIdentifier: {
    type: String,
    required: true,
  },
});

extendAdminInstance({context: {orderIdentifier: props.orderIdentifier}});
</script>
