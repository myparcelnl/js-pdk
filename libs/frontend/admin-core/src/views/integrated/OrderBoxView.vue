<template>
  <OrderBox />
</template>

<script lang="ts" setup>
import OrderBox from '../OrderBox.vue';
import {extendAdminInstance} from '../../composables';

const props = defineProps({
  orderIdentifier: {
    type: String,
    required: true,
  },
});

extendAdminInstance({context: {orderIdentifier: props.orderIdentifier}});
</script>
