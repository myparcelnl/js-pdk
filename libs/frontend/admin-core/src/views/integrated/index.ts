export {default as ModalsView} from '../Modals.vue';

export {default as NotificationsView} from '../Notifications.vue';

export {default as OrderBoxView} from './OrderBoxView.vue';

export {default as OrderListItemView} from './OrderListItemView.vue';

export {default as PluginSettingsView} from '../PluginSettings.vue';

export {default as ProductSettingsView} from '../ProductSettings.vue';
