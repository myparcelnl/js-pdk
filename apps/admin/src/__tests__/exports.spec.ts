import {describe, expect, it} from 'vitest';
import * as integrated from '../integrated';
import * as index from '../index';

describe('exports', () => {
  it('exports from index.ts', () => {
    expect(Object.keys(index)).toMatchSnapshot();
  });

  it('exports from integrated.ts', () => {
    expect(Object.keys(integrated)).toMatchSnapshot();
  });
});
