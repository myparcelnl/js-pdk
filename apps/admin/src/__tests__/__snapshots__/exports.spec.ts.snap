// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`exports > exports from index.ts 1`] = `
[
  "ActionButton",
  "AdminAction",
  "AdminContextKey",
  "AdminIcon",
  "AdminInstanceContextKey",
  "AdminModalKey",
  "LogLevel",
  "NotificationCategory",
  "NotificationContainer",
  "PlainElement",
  "ReturnsForm",
  "ShipmentBarcode",
  "ShipmentOptionsForm",
  "ShipmentStatus",
  "StatusIndicator",
  "SubmitButton",
  "TabNavigation",
  "generateFieldId",
  "getActionIdentifier",
  "modalCloseAction",
  "orderExportAction",
  "orderExportToShipmentsAction",
  "orderViewInBackofficeAction",
  "ordersEditAction",
  "ordersExportPrintShipmentsAction",
  "ordersFetchAction",
  "ordersPrintAction",
  "ordersUpdateAction",
  "shipmentsExportReturnAction",
  "shipmentsPrintAction",
  "shipmentsUpdateAction",
  "useActionStore",
  "useAdminConfig",
  "useAppInfo",
  "useCheckboxGroupContext",
  "useDropOffInputContext",
  "useDropdownData",
  "useElement",
  "useForm",
  "useFormatter",
  "useGlobalContext",
  "useInputWithOptionsContext",
  "useLanguage",
  "useModalContext",
  "useModalElementContext",
  "useModalStore",
  "useNotificationStore",
  "useOrderData",
  "useOrdersData",
  "useProductData",
  "useRadioGroupContext",
  "useSelectInputContext",
  "useTriStateInputContext",
  "useWeekdays",
  "webhooksDeleteAction",
  "AdminComponent",
  "AdminView",
  "BackendEndpoint",
  "Size",
  "Status",
  "Variant",
  "TriStateValue",
  "PdkAdmin",
  "createPdkAdmin",
  "useGlobalPdkAdmin",
]
`;

exports[`exports > exports from integrated.ts 1`] = `
[
  "ActionButton",
  "AdminAction",
  "AdminContextKey",
  "AdminIcon",
  "AdminInstanceContextKey",
  "AdminModalKey",
  "LogLevel",
  "NotificationCategory",
  "NotificationContainer",
  "PlainElement",
  "ReturnsForm",
  "ShipmentBarcode",
  "ShipmentOptionsForm",
  "ShipmentStatus",
  "StatusIndicator",
  "SubmitButton",
  "TabNavigation",
  "generateFieldId",
  "getActionIdentifier",
  "modalCloseAction",
  "orderExportAction",
  "orderExportToShipmentsAction",
  "orderViewInBackofficeAction",
  "ordersEditAction",
  "ordersExportPrintShipmentsAction",
  "ordersFetchAction",
  "ordersPrintAction",
  "ordersUpdateAction",
  "shipmentsExportReturnAction",
  "shipmentsPrintAction",
  "shipmentsUpdateAction",
  "useActionStore",
  "useAdminConfig",
  "useAppInfo",
  "useCheckboxGroupContext",
  "useDropOffInputContext",
  "useDropdownData",
  "useElement",
  "useForm",
  "useFormatter",
  "useGlobalContext",
  "useInputWithOptionsContext",
  "useLanguage",
  "useModalContext",
  "useModalElementContext",
  "useModalStore",
  "useNotificationStore",
  "useOrderData",
  "useOrdersData",
  "useProductData",
  "useRadioGroupContext",
  "useSelectInputContext",
  "useTriStateInputContext",
  "useWeekdays",
  "webhooksDeleteAction",
  "AdminComponent",
  "AdminView",
  "BackendEndpoint",
  "Size",
  "Status",
  "Variant",
  "TriStateValue",
  "ChildProductSettingsView",
  "ModalsView",
  "NotificationsView",
  "OrderBoxView",
  "OrderListItemView",
  "PluginSettingsView",
  "ProductSettingsView",
  "createPdkAdminPlugin",
  "useQueryStore",
  "useStoreQuery",
]
`;
